Some of the recently added tests require sm80

vwbaker opened the pull request:
mma_reduction_layout_analysis.mlir and mma_using_layout_analysis.mlir require gpu sm80 and are failing when we copy them internally without this flag.

vwbaker committed:
Some of the recently added tests require sm80

mma_reduction_layout_analysis.mlir and mma_using_layout_analysis.mlir
require gpu sm80 and are failing when we copy them internally without
this flag.

ScottTodd commented on tests/transform_dialect/cuda/BUILD.bazel:
```diff
@@ -71,7 +71,7 @@ iree_lit_test_suite(
         "notsan",
         "noubsan",
-        "requires-gpu-nvidia",
+        "requires-gpu-sm80",
```
It's not great that we have tests depending on specific device architectures. @ThomasRaoux , can we generalize these tests or find a different way to filter them in the build system?

https://github.com/openxla/iree/blob/60b8ef4aa4bfc1db2e4ecafb6f3c8dc417a4da2a/tests/transform_dialect/cuda/mma_reduction_layout_analysis.mlir#L26-L34

(changing tags here to unblock our infra process is okay though)

ThomasRaoux replied:
I'm not sure what this means to have they filter in the build system? We have quite a few tests here and in e2e/matmul that explicitly use features only available on sm80+. I don't have any opinion on the best way to do it and I'm fine with whatever is preferred but we will need some sets of tests that can only run on those platforms.

GMNGeoffrey replied:
It's fine to have some tests that can only run on a specific architecture when they're arch-specific tests, but we don't want to in general limit our GPU tests to that. Could we separate them out and specifically label them? This is similar to the case of Vulkan fp16

vwbaker replied:
Yes, I separated out the tests that require it.

Also, if it helps, requires-gpu-sm80 will have the tests run on anything sm80+, not just sm80 (there's another tag for that - requires-gpu-sm80-only). So it's not limiting to a _single_ architecture, just the earliest that it can run on.

GMNGeoffrey reviewed (approved):
I think we can merge this and continue the conversation about better management of arch-specific tests.

@vwbaker if you update the CMakeLists.txt file this should be good to go

vwbaker committed:
Separate tests and update cmake

Separate the sm80+ tests from the generic nvidia test and run build_tools/bazel_to_cmake/bazel_to_cmake.py
